addons/sale/models/sale_order_line.py

```diff
@@ -944,15 +944,28 @@ def _compute_qty_to_invoice(self):
         """
         Compute the quantity to invoice. If the invoice policy is order, the quantity to invoice is
         calculated from the ordered quantity. Otherwise, the quantity delivered is used.
+        For combo product lines, first compute all other lines, and then set quantity to invoice
+        only if at least one of its linked lines is invoiceable.
```

Comment:
Nit (to account for optional products):
```suggestion
        only if at least one of its combo item lines is invoiceable.
```